src/machine.rs

```diff
@@ -453,13 +458,147 @@ impl OlmMachine {
         self.inner.cross_signing_status().await.into()
     }
 
+    /// Create a new cross signing identity and get the upload request
+    /// to push the new public keys to the server.
+    ///
+    /// Warning: This will delete any existing cross signing keys that
+    /// might exist on the server and thus will reset the trust
+    /// between all the devices.
+    ///
+    /// Uploading these keys will require user interactive auth.
+    #[napi]
+    pub async fn bootstrap_cross_signing(&self, reset: bool) -> napi::Result<()> {
```

Comment:
Perhaps something like this could explain what the `reset` argument is about, notably missing from the crypto crate docs as well:

```rust
/// # Arguments
///     * `reset` - This argument determines whether the method should create a new
///                 identity or use the existing one during the request. If set to true, the
///                 request will attempt to upload a new identity. If set to false, the request
///                 will attempt to upload the existing identity. Since the uploading process
///                 requires user interactive authentication, which involves sending out the 
///                 same request multiple times, setting this argument to false enables you
///                 to reuse the same request.
```

Reply:
Will do.
Note that this doesn't yet return an array of `OutgoingRequest`s like the WASM bindings do, but that's because there's a lot of missing boilerplate code for the returned types that I would rather get to later in the interest of time.  I do have a work-in-progress change to submit for that eventually.